Assets/Scripts/Interaction Scripts/ControllerState.cs

```diff
@@ -0,0 +1,195 @@
+			LeftAngularVelocity = VRTK_DeviceFinder.GetControllerAngularVelocity(VRTK_DeviceFinder.GetControllerReferenceLeftHand());
+
+            // Receive input information from the right controller.
+			RightIsGrabbing = (rightInteractGrab.GetGrabbedObject()) ? true : false;
+			RightIndex = right.IsButtonPressed(VRTK_ControllerEvents.ButtonAlias.TriggerPress);
+			RightMiddle = right.IsButtonPressed(VRTK_ControllerEvents.ButtonAlias.GripPress);
+			RightA = right.IsButtonPressed(VRTK_ControllerEvents.ButtonAlias.ButtonOnePress);
+			RightB = right.IsButtonPressed(VRTK_ControllerEvents.ButtonAlias.ButtonTwoPress);
+			RightThumb = right.IsButtonPressed(VRTK_ControllerEvents.ButtonAlias.TouchpadPress);
+			RightThumbDelta = right.GetAxis(VRTK_ControllerEvents.Vector2AxisAlias.Touchpad);
+			RightThumbAngle = right.GetAxisAngle(VRTK_ControllerEvents.Vector2AxisAlias.Touchpad);
+			RightOrigin = VRTK_DeviceFinder.GetControllerOrigin(VRTK_DeviceFinder.GetControllerReferenceRightHand());
+			RightVelocity = VRTK_DeviceFinder.GetControllerVelocity(VRTK_DeviceFinder.GetControllerReferenceRightHand());
+			RightAngularVelocity = VRTK_DeviceFinder.GetControllerAngularVelocity(VRTK_DeviceFinder.GetControllerReferenceRightHand());
+
+		}
+
+
+		// Getter methods, left controller.
+		public bool GetLeftIsGrabbing()
+		{
+			return LeftIsGrabbing;
+		}
+		
+		public bool GetLeftIndex()
+		{
+			return LeftIndex;
+		}
+
+		public bool GetLeftMiddle()
+		{
+			return LeftMiddle;
+		}
+		
+		public bool GetLeftX()
+		{
+			return LeftX;
+		}
+
+		public bool GetLeftY()
+		{
+			return LeftY;
+		}
+
+		public bool GetLeftThumb()
+		{
+			return LeftThumb;
+		}
+
+		public Vector2 GetLeftThumbDelta()
+		{
+			return LeftThumbDelta;
+		}
+
+		public float GetLeftThumbAngle()
+		{
+			return LeftThumbAngle;
+		}
+
+		public Transform GetLeftOrigin()
+		{
+			return LeftOrigin;
+		}
+
+		public Vector3 GetLeftVelocity()
+		{
+			return LeftVelocity;
+		}
+
+		public Vector3 GetLeftAngularVelocity()
+		{
+			return LeftAngularVelocity;
+		}
+
+
+		// Getter methods, right controller.
+		public bool GetRightIsGrabbing()
+		{
+			return RightIsGrabbing;
+		}
+		
+		public bool GetRightIndex()
+		{
+			return RightIndex;
+		}
+
+		public bool GetRightMiddle()
+		{
+			return RightMiddle;
+		}
+		
+		public bool GetRightA()
+		{
+			return RightA;
+		}
+
+		public bool GetRightB()
+		{
+			return RightB;
+		}
+
+		public bool GetRightThumb()
+		{
+			return RightThumb;
+		}
+
+		public Vector2 GetRightThumbDelta()
+		{
+			return RightThumbDelta;
+		}
+
+		public float GetRightThumbAngle()
+		{
+			return RightThumbAngle;
+		}
+
+		public Transform GetRightOrigin()
+		{
+			return RightOrigin;
+		}
+
+		public Vector3 GetRightVelocity()
+		{
+			return RightVelocity;
+		}
+
+		public Vector3 GetRightAngularVelocity()
+		{
+			return RightAngularVelocity;
+		}
+
+	}
```

Comment:
check indentation idt this is 4 spaces to a tab.

Reply:
I think GitHub hides the namespace brackets. There's normally one more bracket below.